action/editcommit.php

```diff
@@ -32,23 +32,36 @@ public function register(Doku_Event_Handler $controller) {
         $controller->register_hook('DOKUWIKI_DONE', 'AFTER', $this, 'handle_periodic_pull');
     }
 
-    private function initRepo() {
-        //get path to the repo root (by default DokuWiki's savedir)
-        if(defined('DOKU_FARM')) {
-            $repoPath = $this->getConf('repoPath');
-        } else {
-            $repoPath = DOKU_INC.$this->getConf('repoPath');
-        }
+    private function initRepo($initRepo=True, $filePath="") {
+		if($initRepo) {
+			//get path to the repo root (by default DokuWiki's savedir)
+			if(defined('DOKU_FARM')) {
+				$repoPath = $this->getConf('repoPath');
+			} else {
+				$repoPath = DOKU_INC.$this->getConf('repoPath');
+			}
+		} else {
+			$repoPath = dirname($filePath);
+		}
         //set the path to the git binary
         $gitPath = trim($this->getConf('gitPath'));
         if ($gitPath !== '') {
             Git::set_bin($gitPath);
         }
-        //init the repo and create a new one if it is not present
-        io_mkdir_p($repoPath);
-        $repo = new GitRepo($repoPath, $this, true, true);
+		if ($initRepo) {
+			//init the repo and create a new one if it is not present
+			io_mkdir_p($repoPath);
+			$repo = new GitRepo($repoPath, $this, true, true);
+		} else {
+			new GitRepo($repoPath, $this, false, false);
```

Comment:
It's possible. After using it a little I've seen several issues that need fixing. For instance if one file is not in a git repo, it will currently raise an exception, which is bad. I'll report here the bugs I find